### PR TITLE
DM-28995 Support summit activities

### PR DESCRIPTION
**Pull request by tribeiro**
* Add workaround to edge condition while homing the ATDome.
    If the dome is pressing the home switch and we send a home command, it will simply register the dome as homed and won't send any event to indicate the activity is complete.
  * Add method to reset all offsets in base_tcs.
  * Add set_rem_loglevel method in RemoteGroup, that allows users to set the log level for the remotes loggers.
  * Fix "restore check" feature in prepare for flats.
  * Fix direction of PhysicalSky rotator strategy.
  * Update ATCS to support specifying rotator park position and flat field position.
    When using point_azel to slew the telescope for a safe position, use the current nasmyth position.
  * Fix setting rotFrame in xml7/8 compatibility mode.
  * Update ronchi170lpmm sweet spot.
  * Support differential ra/dec tracking in BaseTCS.

### CODE REVIEW

**Review by patrickingraham — CHANGES_REQUESTED** [review]
Looks good. Just a few minor changes requested.

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 1]
```diff
@@ -817,9 +868,20 @@ async def open_m1_cover(self):
...
             if tel_pos.elevationCalculatedAngle[-1] < self.tel_el_operate_pneumatics:
+
+                try:
+                    nasmyth = await self.rem.atmcs.tel_mount_Nasmyth_Encoders.aget(
+                        timeout=self.fast_timeout
+                    )
+                except asyncio.TimeoutError:
+                    raise RuntimeError("Cannot determine nasmyth position.")
+
+                nasmyth_angle = np.mean(nasmyth.nasmyth2CalculatedAngle)
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/auxtel/atcs.py:879]: So this is going to cause issues if we ever want to put the instrument on the other port...
there are probably others lurking as well. Beyond the scope of this ticket but maybe we need to note somewhere that LATISS must be installed on port 2?

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 2]
```diff
@@ -817,9 +868,20 @@ async def open_m1_cover(self):
             tel_pos = await self.next_telescope_position(timeout=self.fast_timeout)
 
             if tel_pos.elevationCalculatedAngle[-1] < self.tel_el_operate_pneumatics:
+
+                try:
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/auxtel/atcs.py:872]: Why does this try need to be here?

File: python/lsst/ts/observatory/control/base_tcs.py [hunk 3]
```diff
@@ -168,6 +174,10 @@ async def slew_object(
             Default is `SkyAuto`, the rotator is positioned with respect to the
             North axis and is automacally wrapped if outside the limit. See
             `RotType` for more options.
+        dra : `float`
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/base_tcs.py:177]: I don't understand that these are used for. Seems a weird way to do non-sidereal tracking....

File: python/lsst/ts/observatory/control/base_tcs.py [hunk 4]
```diff
@@ -628,6 +661,25 @@ async def offset_xy(self, x, y, relative=False, persistent=False):
             )
             await self.offset_azel(az, el, relative)
 
+    async def reset_offsets(self):
+        """Reset all offsets.
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/base_tcs.py:665]: maybe say "Reset both persistent and relative offsets" 
Then say which methods to use to remove just one, or just the other....
Actually, you could add flags here such that it also has the capability to reset only persistent or only relative?

File: python/lsst/ts/observatory/control/remote_group.py [hunk 5]
```diff
@@ -724,6 +724,20 @@ async def standby(self):
 
         await self.set_state(salobj.State.STANDBY)
 
+    def set_rem_loglevel(self, level):
+        """Set remotes log level.
+
+        Usefull to prevent the internal salobj warnings when read queues are
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/remote_group.py:730]: Useful

File: python/lsst/ts/observatory/control/remote_group.py [hunk 6]
```diff
@@ -724,6 +724,20 @@ async def standby(self):
...
+        filled up.
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/remote_group.py:731]: filling up, no? it warns before full I thought...

**Review by patrickingraham — CHANGES_REQUESTED** [review]

File: python/lsst/ts/observatory/control/base_tcs.py [hunk 7]
```diff
@@ -441,41 +459,56 @@ async def slew(
...
                 else self.RotFrame.TARGET,
+                epoch=epoch,
+                equinox=equinox,
+                parallax=parallax,
+                pmRA=pmRA,
+                pmDec=pmDec,
+                rv=rv,
+                dRA=dRA,
+                dDec=dDec,
+                rotMode=rot_mode if rot_mode is not None else self.RotMode.FIELD,
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/base_tcs.py:481]: There's some documentation about how this is supposed to work as part of the package: https://ts-observatory-control.lsst.io/user-guide/tcs-user-guide-generic.html

I think it needs some updating. I can probably do that as part of DM-29197.

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 8]
```diff
@@ -315,19 +317,21 @@ async def prepare_for_flatfield(self, check=None):
 
         # Creates a copy of check so it can modify it freely to control what
         # needs to be verified at each stage of the process.
-        _check = copy.copy(self.check) if check is None else copy.copy(check)
+        check_bckup = copy.copy(self.check) if check is None else copy.copy(check)
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:320]: Well, check is supposed to be user by users and operators to skip using the particular component. Internally it also uses it some times to skip the component, so I need to keep track of internal usages to restore them later. I probably need to document its usages in more detail.. The feature itself also needs some improvements. 

There's some documentation on the base class: https://ts-observatory-control.lsst.io/py-api/lsst.ts.observatory.control.RemoteGroup.html#remotegroup but definitely need something on the main documentation as well. I would propose to do it in DM-29197.

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 9]
```diff
@@ -635,14 +643,57 @@ async def home_dome(self):
...
+        # we issue a home command.
+        # See DM-29202 for a more permanent fix.
+
+        # Check if the dome is in zero. If it is, assume dome is homed and
+        # move forward.
+        dome_pos = await self.rem.atdome.tel_position.next(
+            flush=True, timeout=self.fast_timeout
         )
+
+        az_position = Angle(dome_pos.azimuthPosition * u.deg).wrap_at("180d")
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:656]: So, if I grab it before sending the home command, the dome could be standing at a random position while not homed that as `azimuthPosition=0`. By grabbing it here, the dome will probably be moving already, reducing the chance of that happening.

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 10]
```diff
@@ -635,14 +643,57 @@ async def home_dome(self):
...
+                # home switch.
+                az_state = await self.rem.atdome.evt_azimuthState.next(
+                    flush=False, timeout=self.fast_timeout
+                )
+            except asyncio.TimeoutError:
+                # The event timedout, this probably means the switch was
+                # pressed. Log a warning and move on.
+                self.log.warning(
+                    "Timeout waiting for ATDome azimuthState event. This may "
+                    "indicate that the dome is already homed and pressing the home switch."
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:674]: done

File: python/lsst/ts/observatory/control/auxtel/atcs.py (hunk 1 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:879]: Yeah, I was lazy when I implemented this the first time. I'll check which one is selected and act accordingly. I am already doing this in other places as well.

File: python/lsst/ts/observatory/control/auxtel/atcs.py (hunk 2 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:872]: Replaced by new code.

File: python/lsst/ts/observatory/control/remote_group.py (hunk 6 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/remote_group.py]: yes

File: python/lsst/ts/observatory/control/remote_group.py (hunk 5 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/remote_group.py]: right

File: python/lsst/ts/observatory/control/base_tcs.py (hunk 3 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/base_tcs.py:177]: Resolved in private

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 11]
```diff
@@ -635,14 +643,57 @@ async def home_dome(self):
...
+            az_state = await self.rem.atdome.evt_azimuthState.next(
+                flush=False, timeout=self.open_dome_shutter_time
+            )
         while az_state.homing:
             self.log.info("Dome azimuth still homing.")
             az_state = await self.rem.atdome.evt_azimuthState.next(
                 flush=False, timeout=self.open_dome_shutter_time
             )
+        else:
+            self.log.info("Dome azimuth hommed successfully.")
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py]: done

File: doc/version_history.rst [hunk 12]
```diff
@@ -9,6 +9,30 @@ Version History
 v0.7.1
 ======
 
+Changes:
+
+  * Add workaround to edge condition while homing the ATDome.
+    If the dome is pressing the home switch and we send a home command, it will simply register the dome as homed and won't send any event to indicate the activity is complete.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply doc/version_history.rst]: fixed

File: python/lsst/ts/observatory/control/base_tcs.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply python/lsst/ts/observatory/control/base_tcs.py:481]: Ok. we need a way to refer to non-docstring docs in docstrings... but that's beyond this ticket.

File: python/lsst/ts/observatory/control/auxtel/atcs.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:656]: no convinced... but not worth squabbling about :)

**Review by patrickingraham — APPROVED** [review]
Looks good. Maybe just add those other changes to your going to do in DM-29197 in the ticket description.
Tests are failing... but jenkins is unhappy so I can't see why...

**Pull request merged**